app/models/manageiq/providers/redhat/infra_manager/ovirt_services/strategies/v3.rb

```diff
@@ -19,5 +25,185 @@ def cluster_name_href(href)
         Ovirt::Cluster.find_by_href(rhevm, href).try(:[], :name)
       end
     end
+
+    # Provisioning
+    def get_host_proxy(host, connection)
+      connection ||= ext_management_system.connect
+      host_proxy = connection.get_resource_by_ems_ref(host.ems_ref)
+      GeneralUpdateMethodNamesDecorator.new(host_proxy)
+    end
+
+    def clone_completed?(args)
+      source = args[:source]
+      phase_context = args[:phase_context]
```

Comment:
I do not know how to do this to be honest (without making a lot of ugly ifs) The phase_context is being modified in a block passed to with_provider_connection which has to be implemented in the appropriate class. Do you have a suggestion how to do it?